### PR TITLE
Reorganize with Package management

### PR DESCRIPTION
**Pull request by willemneal**
With packagement management it makes more sense to break up the library into its constituent parts, even to break out some of it into it's own class.  

From testing using this package as a local dependency things seem to work better, but perhaps we should also add some tests to this package to make sure that with updates to assemblyscript we make sure that package management continues to work.

### CODE REVIEW

**Comment by vgrichina** [comment]
@willemneal will this require updating code using `near.ts`? What is the plan for studio support?

**Comment by bowenwang1996** [comment]
I don't think it's a good idea put the transformer code here (in the source). This repo is for runtime library.

**Comment by willemneal** [comment]
Okay well then I think I'll update the assemblyscript repo so that the only difference is that it will add the transform pass by default.  I have a PR currently on the AS master for changing the types info I need to compile the bindings.  And once that's through we don't need to alter the internals of the near fork anymore and we can just have a different front end.

**Comment by willemneal** [comment]
Ah yeah that way it can also be packaged with webpack too so that we don't even have to worry about finding the file at compile time.

**Comment by bowenwang1996** [comment]
Where do you change the internals now?

**Comment by willemneal** [comment]
The only internals I needed to change are in the PR to upstream.

File: assembly/util.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+export function toString<T>(i: T): string {
+  //@ts-ignore
+  return i.toString();
+}
+// adapted from https://gist.github.com/Juszczak/63e6d9e01decc850de03
+/**
+ * base64 encoding/decoding
+ */
+export namespace base64 {
```
Comment (janedegtiareva) [inline assembly/util.ts]: in runtime_rework branch, I pulled out base64 into a separate class (base64.ts). Do you mind doing that here too to make the merge easier?

Comment (willemneal) [reply assembly/util.ts]: Sure!

**Comment by janedegtiareva** [comment]
This is no longer needed since those changes are in staging branch.

**Pull request closed without merging**